add SLAs to the new Jira field

costaconrado opened the pull request:
This PR adds SLA to the new JIRA Field `SLA Date`.
This PR also renames SLA to SLO to meet correct the naming convention we are using.
This PR also adds a collector to get the new sla_policies file.

Closes OSIDB-4404.

costaconrado commented on apps/sla/models.py:
```diff
@@ -296,23 +296,66 @@ def conditions(self):
         return conditions
 
-    def accepts(self, sla_context):
+    def accepts(self, slo_context):
```
good catch, this can be used with any context, it should be just context, submitting a fix for that.

MrMarble commented on apps/sla/models.py:
```diff
@@ -101,30 +104,28 @@ def parse_type(type_desc):
         for date_source, date_desc_list in date_source_desc.items():
             if date_source not in cls.VALID_DATE_SOURCES:
                 raise SLAExecutionError(
```
Should we rename the exception as well?

costaconrado commented on apps/sla/models.py:
```diff
@@ -334,23 +377,105 @@ def context(self, instance):
+class SLO(TemporalPolicy):
+    """
+    SLO definition and computation model
+    """
```
added the relationship at TemporalPolicy level and removed SLA/SLO models

costaconrado commented on apps/sla/models.py:
```diff
@@ -250,40 +248,42 @@ class SLAPolicy(models.Model):
         return self.name
 
+    @property
+    def temporal_policy(self) -> TemporalPolicy:
```
this method is removed now, it was a mistake since I wasn't using it anywhere.

costaconrado commented:
- fixed the query so SLA does not impact the start date in Jira as it was a mistake

- Exceptions are also renamed

- Renamed PolicyBase to AbstractPolicy, also removed temporal_policy method as it was a leftover from my 1st implementation on this matter and wasn't being used anymore

- I made TemporalPolicy a concrete class and added relationship from SLAPolicy to it

- policy_type renamed to duration_type to better describe what it holds
